### PR TITLE
Upgrade to Patternfly 2.0

### PR DESCRIPTION
**Pull request by erundle**
Updating Angular Patternfly to use patternfly 2.0.

### CODE REVIEW

**Comment by erundle** [comment]
The only lines that changed on the gruntfile are 99-110 which represented additional script and css includes needed for ngDocs. I 100% blew away my fork and started fresh and it is still flagging that file so I don't think there is anything I can do to fix the formatting "flag"

**Comment by waldenraines** [comment]
> I 100% blew away my fork and started fresh and it is still flagging that file so I don't think there is anything I can do to fix the formatting "flag"

Is it possible that it's the line endings that are different?

https://help.github.com/articles/dealing-with-line-endings/

**Comment by erundle** [comment]
Wouldn't all my files be flagged if was a line ending issue? It is picking up whitespace differences but I didn't format the file. You can see the diff without the whitespace changes here:
https://github.com/patternfly/angular-patternfly/pull/24/files?w=1

**Comment by waldenraines** [comment]
My guess is still the line endings.  Perhaps your editor is automatically replacing them?

```
[walden@wraines-sandbox angular-patternfly ((4719e39...))]$ file Gruntfile.js 
Gruntfile.js: ASCII English text, with CRLF line terminators
```

**Comment by waldenraines** [comment]
See also https://github.com/patternfly/angular-patternfly/pull/25.  You may want to wait until that is merged and then rebase in order to get rid of the whitespace changes to `Gruntfile.js`

**Comment by erundle** [comment]
OK, please let me know when your PR is in so I can move forward. I would really like to have the date picker in by EOD if possible.

**Comment by waldenraines** [comment]
> OK, please let me know when your PR is in so I can move forward. I would really like to have the date picker in by EOD if possible.

I don't have merge capabilities for angular-patternfly so hopefully someone can merge this for me.

**Comment by erundle** [comment]
I will go ahead and accept it. Thanks for your help on this.

**Comment by erundle** [comment]
Merged in PR 25 and pushed. The line ending changes did the trick.

**Comment by waldenraines** [comment]
What about rebasing after #25 instead of merging it in?  That way this PR will only have your changes and thus be easier to review.

**Comment by erundle** [comment]
Ok, I rolled back the merge and rebased instead.

**Comment by erundle** [comment]
Does anyone have any issues with this? Or can this be pulled in?

File: Gruntfile.js [hunk 1]
```diff
@@ -96,14 +96,18 @@ module.exports = function(grunt) {
...
           dest: 'dist/docs',
-          scripts: ['lib/patternfly/components/jquery/jquery.js',
+          scripts: ['lib/patternfly/components/jquery/dist/jquery.js',
             'lib/patternfly/components/bootstrap/dist/js/bootstrap.js',
-            'lib/patternfly/components/bootstrap-select/bootstrap-select.js',
+            'lib/patternfly/components/bootstrap-combobox/js/bootstrap-combobox.js',
+            'lib/patternfly/components/bootstrap-select/js/bootstrap-select.js',
+            'lib/patternfly/components/bootstrap-treeview/src/js/bootstrap-treeview.js',
+            'lib/patternfly/components/c3/c3.js',
+            'lib/patternfly/components/d3/d3.js',
```
Comment (waldenraines) [inline Gruntfile.js]: c3 and d3 don't appear related to this upgrade.

File: misc/bootstrap-overrides.css [hunk 2]
```diff
@@ -0,0 +1,7 @@
+/* This is needed since ngDocs has bootstrap as a default file to load
+ * patternfly assumes that the consuming application is not loading the bootstrap.css directly
+  */
```
Comment (waldenraines) [inline misc/bootstrap-overrides.css]: Any chance you can explain this a little more?  I don't follow.

File: Gruntfile.js (hunk 1 above)
Comment (erundle) [inline Gruntfile.js]: These were added in PF 1.3.0 so we are playing catchup with the includes since this project hasn't been upgraded in a bit. Combobox and treeview were also added a bit back but.

Granted you don't need them to run angular-pf as of yet but it if you want the docs to explain what includes PF needs then it is nice to have them here. So from my perspective it is a nice to have.

File: misc/bootstrap-overrides.css (hunk 2 above)
Comment (erundle) [inline misc/bootstrap-overrides.css]: Sure, in ngDocs index.tmpl they have the following line: addTag('link', {rel: 'stylesheet', href: 'css/bootstrap.min.css', type: 'text/css'});

This pulls in the default out of box bootstrap.css. The patternfly.css is meant to be installed as a replacement for this file not as a stand aside. So this was causing conflicts with the form grids. If you remove that style you will see on all the examples that use a 12 column grid the labels and fields are not on the same line. This fixes that issue by addressing the style clash.

File: Gruntfile.js (hunk 1 above)
Comment (waldenraines) [inline Gruntfile.js]: :+1:

**Comment by waldenraines** [comment]
LGTM, thanks.
